hera_sim/visibilities/fftvis.py

```diff
@@ -0,0 +1,318 @@
+            raise ValueError(
+                """
+                While UVData allows non-linear polarizations, they are not suitable
+                for generating simulations. Please convert your UVData object to use
+                linear polarizations before simulating (and convert back to
+                other polarizations afterwards if necessary).
+                """
+            )
+
+        do_pol = self._check_if_polarized(data_model)
+        if do_pol:
+            # Number of feeds must be two if doing polarized
+            try:
+                nfeeds = uvbeam.data_array.shape[1 if uvbeam.future_array_shapes else 2]
+            except AttributeError:
+                # TODO: the following assumes that analytic beams are 2 feeds unless
+                # otherwise specified. This should be fixed at the AnalyticBeam API
+                # level.
+                nfeeds = getattr(uvbeam, "Nfeeds", 2)
+
+            assert nfeeds == 2
+
+    def estimate_memory(self, data_model: ModelData) -> float:
+        """
+        Estimates the memory usage of the model.
+
+        Parameters
+        ----------
+        data_model : ModelData
+            The model data.
+
+        Returns
+        -------
+        float
+            Estimated memory usage in GB.
+        """
+        bm = data_model.beams[0]
+        nt = len(data_model.lsts)
+        nax = getattr(bm, "Naxes_vec", 1)
+        nfd = getattr(bm, "Nfeeds", 1)
+        nant = len(data_model.uvdata.antenna_names)
+        nsrc = len(data_model.sky_model.ra)
+        nbeam = len(data_model.beams)
+        nf = len(data_model.freqs)
+
+        # Estimate size of the FFT grid used to compute the visibilities
+        active_antpos_array, _ = data_model.uvdata.get_ENU_antpos(pick_data_ants=True)
+        # Estimate the size of the grid used to compute the visibilities
+        max_blx, max_bly, _ = np.abs(
+            active_antpos_array.max(axis=0) - active_antpos_array.min(axis=0)
+        )
+        avg_freq = np.mean(data_model.freqs)
+        n_gridx = int(8 * avg_freq * max_blx / 3e8)  # number of grid points in u/l axis
+        n_gridy = int(8 * avg_freq * max_bly / 3e8)  # number of grid points in v/m axis
+
+        try:
+            nbmpix = bm.data_array[..., 0, :].size
+        except AttributeError:
+            nbmpix = 0
+
+        all_floats = (
+            nf * nt * nfd**2 * nant**2  # visibilities
+            + n_gridx * n_gridy  # FFT grid
+            + nf * nbeam * nbmpix  # raw beam
+            + nax * nfd * nbeam * nsrc / 2  # interpolated beam
+            + 3 * nant  # antenna positions
+            + nsrc * nf  # source fluxes
+            + nt * 9  # rotation matrices
+            + 3 * nsrc
+            + 3 * nsrc  # source positions (topo and eq)
+        )
+
+        return all_floats * self._precision * 4 / 1024**3
+
+    def simulate(self, data_model):
+        """
+        Calls :func:fftvis to perform the visibility calculation.
```

Comment:
After :func:, the fftvis should be surrounded in backticks

Reply:
Fixed this